sys/parsoid.js

```diff
@@ -318,7 +318,8 @@ class ParsoidService {
                             // This revision is actually a redirect. Pass redirect target
                             // to caller, and let it rewrite the location header.
                             resp.status = 302;
-                            resp.headers.location = encodeURIComponent(redirectTarget);
+                            // Filter will take care of uri-encoding
+                            resp.headers.location = redirectTarget;
```

Comment:
Could the decoding be done once in the filter & the result be stored in a local variable? The performance impact should be pretty small, and in exchange we'd avoid introducing an exception to the "regular HTTP headers" principle & assumptions about which filters are active.